### PR TITLE
fix(tx): guard parallel writes to local result

### PR DESCRIPTION
**Pull request by dranikpg**
Guards access to local_result_ in case of oom error where multiple shards might write into it at once -> ub

### CODE REVIEW

File: src/server/transaction.cc [hunk 1]
```diff
@@ -484,6 +484,8 @@ bool Transaction::RunInShard(EngineShard* shard, bool txq_ooo) {
       local_result_ = status;
     } else {
       if (status == OpStatus::OUT_OF_MEMORY) {
+        absl::base_internal::SpinLockHolder lk{&local_result_mu_};
+        CHECK(local_result_ == OpStatus::OK || local_result_ == OpStatus::OUT_OF_MEMORY);
```
Comment (adiholden) [inline src/server/transaction.cc:488]: I  believe we have the same issue below inside catch

Comment (dranikpg) [reply src/server/transaction.cc:488]: True, fixed

File: src/server/transaction.cc [hunk 2]
```diff
@@ -484,14 +484,16 @@ bool Transaction::RunInShard(EngineShard* shard, bool txq_ooo) {
...
         local_result_ = status;
       } else {
         CHECK_EQ(OpStatus::OK, status);
       }
     }
   } catch (std::bad_alloc&) {
-    // TODO: to log at most once per sec.
-    LOG_FIRST_N(ERROR, 16) << " out of memory";
+    LOG_FIRST_N(ERROR, 16) << " out of memory";  // TODO: to log at most once per sec.
```
Comment (chakaz) [inline src/server/transaction.cc:495]: How about `LOG_EVERY_N_SEC(ERROR, 600)` (or any other value)?

**Review by adiholden — APPROVED** [review]

File: src/server/transaction.cc (hunk 2 above)
Comment (dranikpg) [reply src/server/transaction.cc:495]: We don't have it for some reason, but upstream absl has it 🤔 I guess we need to update deps

Comment (chakaz) [reply src/server/transaction.cc:495]: It might be because we're using glog directly and not through absl, at least that's something I remember seeing?

Comment (dranikpg) [reply src/server/transaction.cc:495]: Yes, let's fix it in a separate PR

Comment (romange) [reply src/server/transaction.cc:495]: no, no please do not fix it.

**Pull request merged**